addons/account/models/account_move.py

```diff
@@ -1878,11 +1896,23 @@ def _get_tax_totals(self, partner, tax_lines_data, amount_total, amount_untaxed,
     @api.depends('date', 'line_ids.debit', 'line_ids.credit', 'line_ids.tax_line_id', 'line_ids.tax_ids', 'line_ids.tax_tag_ids')
     def _compute_tax_lock_date_message(self):
         for move in self:
-            if move._affect_tax_report() and move.company_id.tax_lock_date and move.date and move.date <= move.company_id.tax_lock_date:
-                move.tax_lock_date_message = _(
-                    "The accounting date is set prior to the tax lock date which is set on %s. "
-                    "Hence, the accounting date will be changed to %s.",
-                    format_date(self.env, move.company_id.tax_lock_date), format_date(self.env, fields.Date.context_today(self)))
+            invoice_date = move.invoice_date or fields.Date.context_today(move)
+            accounting_date = move.date or fields.Date.context_today(move)
+            lock_dates = move._get_violated_lock_dates(accounting_date, move._affect_tax_report())
+            if lock_dates:
+                accounting_date = move._get_accounting_date(invoice_date, move._affect_tax_report())
+                lock_date, lock_type = lock_dates[-1]
+                tax_lock_date_message = _(
+                    "The accounting date being set prior to the %(lock_type)s lock date %(lock_date)s, it will be changed to %(accounting_date)s upon posting.",
```

Comment:
Keep lines shorter
```suggestion
                    "The accounting date being set prior to the %(lock_type)s lock date %(lock_date)s,"
                    " it will be changed to %(accounting_date)s upon posting.",